### DISCUSSION TITLE
Add open state to lock

### DISCUSSION THREAD

**Discussion by gjohansson-ST** [home-assistant/architecture#774] [Entity Models]
From discussion in https://community.home-assistant.io/t/support-3-states-for-lock-template/423831

I think this text explains it pretty well:
"The lock can be locked or unlocked. However, [I learned some while ago 1](https://community.home-assistant.io/t/mqtt-lock-open/232823/10) that certain European locks can have an “open” state where both their deadbolt and striker are retracted thereby allowing the door to be opened without having to turn the door handle."

I think it would make sense to implement this additional state for locks.

**Comment by bdraco** [comment]
When considering changes to the model, we ask for a list of integrations that could use this new state so we can evaluate the tradeoff of having to support the additional complexity.

**Reply by gjohansson-ST** [reply]
Good question. I can implement it myself for `yale_smart_alarm` so I would suppose maybe it's also an available state for `august` and given the conversation I linked too in the original post it should be implemented for `mqtt`.
If this is possible for other lock integrations I can only hope other codeowners can answer that.

**Comment by Nardol** [comment]
Could a opening state also be considered in this discussion or should I open a new one?
Because it could be useful to know the lock is open (or unlatched) but also if the action is occurring as it is the case for lock and unlock?

I am not a code owner but I could implement it to MQTT as I did for locking/unlocking.

**Comment by jbouwh** [comment]
This is not just a new state. The `lock` state is determined by properties. So we possible would need an `is_open` entity property as well. Open should what priority concerns be handled before `unlocked` and may be even above `jammed`.

**Reply by gjohansson-ST** [reply]
That would be the proposal to implement such property.
In terms of priority it should go first as if it's open it's not any of the other states.

We can implement for MQTT as already mentioned.
Yale lock can have such state.
Also probably all locks that goes into doors where you can leave the handle open could be implement.

**Reply by jbouwh** [reply]
I would not be against, there is even a feature and service call to open the lock, so it is somehow weird the state `open` is missing,
I would suggest to add property `is_open` as well, for the active state I believe `open` should go before `jammed`.

**Reply by gjohansson-ST** [reply]
@jbouwh I wrote my proposal as a new comment below. Do you think it's feasible?
I can do the PR if you agree and we provide it as the answer.

**Comment by gjohansson-ST** [comment]
### Proposal

Expand the current `lock` entity component to include

**Two new states**
- `opening`
- `open`

**Two new properties**
- `is_opening`
- `is_opened`

function `open` already exist so no addition

**Active state**
```python
if self.is_opened:
    return STATE_OPENED
if self.is_jammed:
    return STATE_JAMMED
if self.is_locking:
    return STATE_LOCKING
if self.is_opening:
    return STATE_OPENING
if self.is_unlocking:
    return STATE_UNLOCKING
if (locked := self.is_locked) is None:
    return None
return STATE_LOCKED if locked else STATE_UNLOCKED
```

**Reply by jbouwh** [reply]
I agree on is `is_opened` and it should have priority on `is_unlocked`. I doubt on `is_opening` as usually this not something that takes much time.

**Reply by gjohansson-ST** [reply]
What is you reference on opening. If I use my yale-lock at home as reference the `opening` state would not be a second even before it's `opened`

**Reply by Nardol** [reply]
If I understand correctly, both of you says the same :slightly_smiling_face: and I also agree, opening is a short transitioning state after unlocked and before opened.

**Reply by emontnemery** [reply]
The proposal was discussed in the core team, the decision is to approve it.
An alternative could be to add a binary sensor to signal the state of the striker, but since there's already a service to open locks it makes perfect sense to also have a corresponding state 👍

**Comment by gjohansson-ST** [comment]
Implementation being done in https://github.com/home-assistant/core/pull/111968

**Comment by emontnemery** [comment]
A question was brought up about which core integrations can make use of the new state. The list below lists integrations which flag support for the OPEN service. I've excluded demo, group and kitchen_sink from the list.

esphome: DIY integration, will support what the entity model supports.
homematic: Open state does not seem to be supported: https://github.com/danielperna84/pyhomematic/blob/8aaf60d6f6ec49d1725702109736d5a0e97d84c9/pyhomematic/devicetypes/actors.py#L554
homematicip: Open state seems to be supported, but not an opening state: https://github.com/hahn-th/homematicip-rest-api/blob/203a987de81df283bbf15f00c141d8460f48a418/src/homematicip/base/enums.py#L640C1-L644C18
loqed: Open state supported, unclear if opening state is also supported: https://support.loqed.com/en/articles/6127856-loqed-local-bridge-api-integration
matter: Open and opening supported
mqtt: DIY integration, will support what the entity model supports.
nuki: Both open and opening states supported: https://developer.nuki.io/page/nuki-bridge-http-api-1-11/4/#heading--lock-states
switchbot: Open state seems to be supported, but not an opening state: https://github.com/Danielhiversen/pySwitchbot/blob/343f6ccecb357245316aa3d7508372e8f8010006/switchbot/const.py#L45-L52
tedee: I believe this is the open state (the latched is called pullspring elsewhere in the library): https://github.com/zweckj/pytedee_async/blob/5b97d336959ef125262a221887175df0046f9631/pytedee_async/lock.py#L15

**Reply by gjohansson-ST** [reply]
We can also add Yale to the list. Not sure about `opening` (but believe so) but definitely `open`.

**Reply by emontnemery** [reply]
@gjohansson-ST is that a core integration?

**Reply by gjohansson-ST** [reply]
Yes. `yale_smart_alarm` and as it's the same brand/locks as `yalexs_ble` and `august` I think it includes them as well.
Probably @bdraco can confirm/deny that.

Don't have a Yale lock anymore but had at the time via `yale_smart_alarm` hence why I opened this discussion as I fell short on this state (knowing that most European locks has support for this at least).

**Reply by gjohansson-ST** [reply]
`verisure` also implements Yale locks

**Reply by emontnemery** [reply]
OK, I see. None of the integrations you mentioned which interact with yale locks support the open service, so I didn't check them or their supporting libraries.
